sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStore.scala

```diff
@@ -369,6 +369,21 @@ trait StateStoreProvider {
   /** Return an instance of [[StateStore]] representing state data of the given version */
   def getStore(version: Long): StateStore
 
+  /**
+   * This is an optional method, used by snapshotStartBatchId option when reading state generated
+   * by join operation as data source.
+   * Return an instance of [[StateStore]] representing state data of the given version.
+   * The State Store will be constructed from the batch at startVersion, and applying delta files
+   * up to the endVersion. If there is no snapshot file of batch startVersion, an exception will
+   * be thrown.
+   *
+   * @param startVersion checkpoint version of the snapshot to start with
+   * @param endVersion checkpoint version to end with
+   */
+  def getStore(startVersion: Long, endVersion: Long): StateStore =
+    throw new SparkUnsupportedOperationException("getStore with startVersion and endVersion " +
```

Comment:
can we just put nothing here? like 
```
def getStore(version: Long): StateStore
```

Reply:
Hmm - what error do you see here ? can you paste it please ?

Reply:
Building on the assumption that when users create custom state store provider and they do not implement this method because it is optional. They will see errors like
```
Missing implementation for member of trait StateStoreProvider
```